commands/main_test.go

```diff
@@ -0,0 +1,223 @@
+package commands_test
+
+import (
+	"database/sql"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/bwmarrin/discordgo"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	testChannelID      = "123456789"
+	testJoke1          = "A steak pun is a rare medium well done."
+	testJoke2          = "What do you call a fish with no eyes?"
+	testJoke2Punchline = "Fsh"
+	testFile1          = "testdata/file1.png"
+	testFile2          = "testdata/file2.png"
+)
+
+var (
+	db *sql.DB
+)
+
+func setup() (err error) {
+	db, err = sql.Open("sqlite3", ":memory:")
+	if err != nil {
+		return
+	}
+
+	_, err = db.Exec(fmt.Sprintf(`
+CREATE TABLE "jokes" (
+  "id"        INTEGER NOT NULL,
+  "text"      TEXT NOT NULL,
+  "punchline" TEXT,
+  PRIMARY KEY("id" AUTOINCREMENT)
+);
+CREATE TABLE "images" (
+	"id"   INTEGER NOT NULL,
+	"path" TEXT NOT NULL,
+	PRIMARY KEY("id" AUTOINCREMENT)
+  );
+CREATE TABLE "taunts" (
+  "id"   INTEGER NOT NULL,
+  "path" TEXT NOT NULL,
+  PRIMARY KEY("id" AUTOINCREMENT)
+);
+
+INSERT INTO "jokes" ("text", "punchline") VALUES
+  (%q, null),
+  (%q, %q);
+INSERT INTO "images" ("path") VALUES
+  (%[4]q),
+  (%[5]q);
+INSERT INTO "taunts" ("path") VALUES
+  (%[4]q),
+  (%[5]q);
+`, testJoke1, testJoke2, testJoke2Punchline, testFile1, testFile2))
+	return
+}
+
+func TestMain(m *testing.M) {
+	if err := setup(); err != nil {
+		log.Fatalln(err)
+	}
+	os.Exit(m.Run())
+}
+
+// Creates a Message with the provided arguments.
+//  0: Content
+//  1: ChannelID
+//  2: MessageID
+func newMsg(args ...string) *discordgo.Message {
+	m := new(discordgo.Message)
+	if len(args) > 0 {
+		m.Content = args[0]
+	}
+	if len(args) > 1 {
+		m.ChannelID = args[1]
+	}
+	if len(args) > 2 {
+		m.ID = args[2]
+	}
+	return m
+}
+
+// Creates a MessageCreate with the provided arguments. Arguments are the same as newMsg.
+func newMsgCreate(args ...string) *discordgo.MessageCreate {
+	return &discordgo.MessageCreate{Message: newMsg(args...)}
+}
+
+func newTestSession() *testsession {
+	return &testsession{
+		messages: make(map[string]*strings.Builder),
+		files:    make(map[string]map[string][]byte),
+	}
+}
+
+type testsession struct {
+	counter      int
+	messages     map[string]*strings.Builder
+	files        map[string]map[string][]byte
+	statusListen string
+	statusPlay   string
+}
+
+func (s *testsession) AddHandler(handler interface{}) func() {
+	return func() {}
+}
+
+func (s *testsession) Channel(channelID string) (*discordgo.Channel, error) {
```

Comment:
WDYT about removing names from unused parameters in these stub implementations?

```suggestion
func (s *testsession) Channel(string) (*discordgo.Channel, error) {
```
I couldn't find any mention of it as a suggestion in your style guide (https://go.dev/doc/effective_go), closest being about using the blank identifier for unused vars.

IIUC one of the reasons golang has support for unnamed identifier is for exactly this case, where you are bringing along some args to satisfy some interface, but don't actually care about those args.

Reply:
Fair point. I'll make that change.

Reply:
Done.